src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -183,20 +192,53 @@ private bool VerifyCSharpPlugin(IProgressController controller, CancellationToke
                 return false;
             }
 
-            var csPlugin = plugins.FirstOrDefault(x => StringComparer.Ordinal.Equals(x.Key, ServerPlugin.CSharpPluginKey));
-            if (string.IsNullOrWhiteSpace(csPlugin?.Version) || VersionHelper.Compare(csPlugin.Version, ServerPlugin.CSharpPluginMinimumVersion) < 0)
-            {
-                string errorMessage = string.Format(CultureInfo.CurrentCulture, Strings.ServerDoesNotHaveCorrectVersionOfCSharpPlugin, ServerPlugin.CSharpPluginMinimumVersion);
+            IsCSharpPluginSupported = VerifyPluginSupport(plugins, MinimumSupportedServerPlugin.CSharp);
+            IsVBNetPluginSupported = VerifyPluginSupport(plugins, MinimumSupportedServerPlugin.VbNet);
 
-                this.host.ActiveSection?.UserNotifications?.ShowNotificationError(errorMessage, NotificationIds.BadServerPluginId, null);
-                notifications.ProgressChanged(errorMessage);
-                notifications.ProgressChanged(Strings.ConnectionResultFailure);
+            var projects = this.projectSystem.GetSolutionProjects().ToList();
+            var anyCSharpProject = projects.Any(MinimumSupportedServerPlugin.CSharp.ISupported);
```

Comment:
I tend to not simplify LINQ this much. I think this is harder to read than `.Any(p => ...IsSupported(p))`. But it's a matter of taste only.